src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -1819,34 +1829,30 @@ int wm_vuldet_linux_rm_false_positivies(sqlite3 *db, agent_software *agent, OSHa
 
         if (agent->dist != FEED_MAC) {
             // We will discard the packages unaffected or already fixed by the vendor (Ubuntu/Debian/RedHat)
-            if (wm_vuldet_linux_rm_nvd_not_affected_packages(db, agent, cve, first, vuln_discarded)) {
-                os_free(vuln_discarded);
+            if (wm_vuldet_linux_rm_nvd_not_affected_packages(db, agent, cve, first, &vuln_discarded)) {
                 return OS_INVALID;
             }
         }
 
         // We will discard the packages that depend on packages not installed
-        wm_vuldet_linux_rm_nvd_not_dependencies_met_packages(cve, first, vuln_discarded);
+        wm_vuldet_linux_rm_nvd_not_dependencies_met_packages(cve, first, &vuln_discarded);
 
         // We will discard the packages that have vulnerable field set to 0
-        wm_vuldet_linux_rm_nvd_not_vulnerable_packages(cve, first, vuln_discarded);
+        wm_vuldet_linux_rm_nvd_not_vulnerable_packages(cve, first, &vuln_discarded);
 
         if (agent->dist != FEED_MAC) {
             // We will discard the packages not fixed by the OVAL that the NVD said they are not affected
-            if (wm_vuldet_linux_rm_oval_not_affected_packages(db, cve, first, vuln_discarded)) {
-                os_free(vuln_discarded);
+            if (wm_vuldet_linux_rm_oval_not_affected_packages(db, cve, first, &vuln_discarded)) {
                 return OS_INVALID;
             }
         }
 
         hash_node = OSHash_Next(cve_table, &inode_it, hash_node);
     }
 
-    mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_VULN_FILTER_AG, *vuln_discarded, atoi(agent->agent_id));
+    mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_VULN_FILTER_AG, vuln_discarded, atoi(agent->agent_id));
```

Comment:
Should we use `scan_ctx->agent_id` here ?

Reply:
I consider `scan_ctx->agent_id` better than calculating the ID with atoi, but this method doesn't have scan_ctx as a parameter. And handling this parameter only for log purposes will not be a good practice.

Reply:
You are right !